sound/soc/sof/sof-priv.h

```diff
@@ -183,6 +183,7 @@ struct snd_sof_dsp_ops {
 	int (*load_firmware)(struct snd_sof_dev *sof_dev); /* mandatory */
 	int (*load_module)(struct snd_sof_dev *sof_dev,
 			   struct snd_sof_mod_hdr *hdr); /* optional */
+	int (*load_library)(struct snd_sof_dev *sdev, struct firmware *fw, u32 id); /* optional */
```

Comment:
what is the difference between `load_module` and `load_library`?

Would we want to unload to make it dynamic?

Reply:
this is a question for @plbossart. I'm not sure about load_module.

Reply:
I have no idea @ujfalusi @ranj063, I don't think anyone ever used this, we always use the default, don't we?

````c
        load_module = sof_ops(sdev)->load_module;
	if (!load_module) {
		dev_dbg(sdev->dev, "%s: Using generic module loading\n", __func__);
		load_module = sof_ipc3_parse_module_memcpy;
	} else {
		dev_dbg(sdev->dev, "%s: Using custom module loading\n", __func__);
	}
 [...]
	ret = load_module(sdev, module);
````

@lgirdwood any memories of what the intent was on this one?

Reply:
I got the answer from Cezary. Apparently in older platform the library and the module code were separate but newer ones CNL onwards dont have that.

Reply:
@plbossart, only the default load_module (`sof_ipc3_parse_module_memcpy()`) is used, all architectures dropped setting the sdev level load_module, all is internal to ipc3 now.